mono/metadata/threads.c

```diff
@@ -6775,3 +6775,16 @@ ves_icall_System_Threading_Thread_GetCurrentProcessorNumber (MonoError *error)
 }
 
 #endif
+
+static
+void mono_thread_attach_debugger (gpointer key, gpointer value, gpointer user)
+{
+	MonoInternalThread* val = (MonoInternalThread*) value;
+	MONO_PROFILER_RAISE (thread_attached, ((uintptr_t) key, val, mono_thread_current_for_thread((MonoInternalThread*) value)));
+}
+
+void 
+mono_threads_attach_debugger ()
+{
+	mono_g_hash_table_foreach (threads, mono_thread_attach_debugger, NULL);
```

Comment:
This needs some locking. Also, it calls a lot of code, which might do its own locking. So it might be better to collect the threads into an array, i.e. see collect_threads (), then process that.